### PR TITLE
Test EventSerializer

### PR DESCRIPTION
**Pull request by andreaTP**
This is more preparation work for bigger refactorings.

### CODE REVIEW

**Review by johanandren — APPROVED** [review]
LGTM except the CI-failing part ;)

**Comment by andreaTP** [comment]
oooops, fixed!

File: sdk/test/effect-serializer-test.js [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+
+    // Act
+    const res = () =>
+      es.serializeEffect(exampleServiceTwo.methods.DoSomethingOne, {}, {});
+
+    // Assert
+    should.throw(() => res(), Error);
+  });
+
+  it('should serialize succesfully', () => {
```
Comment (pvlugter) [inline sdk/test/effect-serializer-test.js:55]: Not important in test names, but:

```suggestion
  it('should serialize successfully', () => {
```

File: sdk/test/effect-serializer-test.js [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+    // Act
+    const res = es.serializeEffect(exampleService.methods.DoSomething, msg, {});
+
+    // Assert
+    res.serviceName.should.eq('com.example.ExampleService');
+    res.commandName.should.eq('DoSomething');
+    res.payload.type_url.should.eq('type.googleapis.com/com.example.In');
+  });
+
+  it('should serialize succesfully', () => {
```
Comment (pvlugter) [inline sdk/test/effect-serializer-test.js:71]:
```suggestion
  it('should serialize successfully', () => {
```

File: sdk/test/effect-serializer-test.js [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+      {},
+    );
+
+    // Assert
+    res.serviceName.should.eq('com.example.ExampleService');
+    res.commandName.should.eq('DoSomething');
+    res.payload.type_url.should.eq('type.googleapis.com/com.example.In');
+  });
+
+  it('should serialize succesfully unresolved methods', () => {
```
Comment (pvlugter) [inline sdk/test/effect-serializer-test.js:91]:
```suggestion
  it('should serialize successfully unresolved methods', () => {
```

File: sdk/test/effect-serializer-test.js [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+    // Act
+    const res = es.serializeEffect(exampleService.methods.DoSomething, msg, {});
+
+    // Assert
+    res.serviceName.should.eq('com.example.ExampleService');
+    res.commandName.should.eq('DoSomething');
+    res.payload.type_url.should.eq('type.googleapis.com/com.example.In');
+  });
+
+  it('should serialize succesfully using lookup', () => {
```
Comment (pvlugter) [inline sdk/test/effect-serializer-test.js:107]:
```suggestion
  it('should serialize successfully using lookup', () => {
```

File: sdk/test/effect-serializer-test.js [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+        exampleServiceGenerated.ExampleServiceTwoService.doSomethingOne,
+        msg,
+        {},
+      );
+
+    // Assert
+    should.throw(() => res(), Error);
+  });
+
+  it('should serialize succesfully methods using the generated gRPC definition', () => {
```
Comment (pvlugter) [inline sdk/test/effect-serializer-test.js:146]:
```suggestion
  it('should serialize successfully methods using the generated gRPC definition', () => {
```

**Review by pvlugter — APPROVED** [review]

**Review by johanandren — APPROVED** [review]

**Comment by andreaTP** [comment]
thanks, @pvlugter for fixing my typo copy-pasted all around 😅

**Pull request merged**